File: RQ2_Prompt_Elements/OpenAI_Data/HumanEvalPython_output/scenario1/id_136T2000Test.py
# id_136_Test.java
from scenario1.id_136 import largest_smallest_integers

import pytest

class id_136_Test:
    """
    Test class of id_136.py.
    It contains ten unit test cases for the :func:`scenario1.id_136.largest_smallest_integers`.
    """


    def test_largest_smallest_integers_1(self):
        """
        Test case 1 for :func:`scenario1.id_136.largest_smallest_integers`.
        """
        assert largest_smallest_integers([-1, -2, -3, -4, -5]) == (-1, None)

    def test_largest_smallest_integers_2(self):
        """
        Test case 2 for :func:`scenario1.id_136.largest_smallest_integers`.
        """
        assert largest_smallest_integers([1, 2, 3, 4, 5]) == (None, 1)

    def test_largest_smallest_integers_3(self):
        """
        Test case 3 for :func:`scenario1.id_136.largest_smallest_integers`.
        """
        assert largest_smallest_integers([-1, 2, -3, 4, -5]) == (-1, 2)

    def test_largest_smallest_integers_4(self):
        """
        Test case 4 for :func:`scenario1.id_136.largest_smallest_integers`.
        """
        assert largest_smallest_integers([-1, -2, -3, -4, -5, 1, 2, 3, 4, 5]) == (-1, 1)

    def test_largest_smallest_integers_5(self):
        """
        Test case 5 for :func:`scenario1.id_136.largest_smallest_integers`.
        """
        assert largest_smallest_integers([-1, -2, -3, -4, -5, 1, 2, 3, 4, 5, -1, -2, -3, -4, -5, 1, 2, 3, 4, 5]) == (-1, 1)

    def test_largest_smallest_integers_6(self):
        """
        Test case 6 for :func:`scenario1.id_136.largest_smallest_integers`.
        """
        assert largest_smallest_integers([-1, -2, -3, -4, -5, 1, 2, 3, 4, 5, -1, -2, -3, -4, -5, 1, 2, 3, 4, 5, -1, -2, -3, -4, -5, 1, 2, 3, 4, 5]) == (-1, 1)

    def test_largest_smallest_integers_7(self):
        """
        Test case 7 for :func:`scenario1.id_136.largest_smallest_integers`.
        """
        assert largest_smallest_integers([-1, -2, -3, -4, -5, 1, 2, 3, 4, 5, -1, -2, -3, -4, -5, 1, 2, 3, 4, 5, -1, -2, -3, -4, -5, 1, 2, 3, 4, 5, -1, -2, -3, -4, -5, 1, 2, 3, 4, 5]) == (-1, 1)

    def test_largest_smallest_integers_8(self):
        """
        Test case 8 for :func:`scenario1.id_136.largest_smallest_integers`.
        """
        assert largest_smallest_integers([-1, -2, -3, -4, -5, 1, 2, 3, 4, 5, -1, -2, -3, -4, -5, 1, 2, 3, 4, 5, -1, -2, -3, -4, -5, 1, 2, 3, 4, 5, -1, -2, -3, -4, -5, 1, 2, 3, 4, 5, -1, -2, -3, -4, -5, 1, 2, 3, 4, 5]) == (-1, 1)

    def test_largest_smallest_integers_9(self):
        """
        Test case 9 for :func:`scenario1.id_136.largest_smallest_integers`.
        """
        assert largest_smallest_integers([-1, -2, -3, -4, -5, 1, 2, 3, 4, 5, -1, -2, -3, -4, -5, 1, 2, 3, 4, 5, -1, -2, -3, -4, -5, 1, 2, 3, 4, 5, -1, -2, -3, -4, -5, 1, 2, 3, 4, 5, -1, -2, -3, -4, -5, 1, 2, 3, 4, 5, -1, -2, -3, -4, -5, 1, 2, 3, 4, 5]) == (-1, 1)

    def test_largest_smallest_integers_10(self):
        """
        Test case 10 for :func:`scenario1.id_136.largest_smallest_integers`.
        """
        assert largest_smallest_integers([-1, -2, -3, -4, -5, 1, 2, 3, 4, 5, -1, -2, -3, -4, -5, 1, 2, 3, 4, 5, -1, -2, -3, -4, -5, 1, 2, 3, 4, 5, -1, -2, -3, -4, -5, 1, 2, 3, 4, 5, -1, -2, -3, -4, -5, 1, 2, 3, 4, 5, -1, -2, -3, -4, -5, 1, 2, 3, 4, 5, -1, -2, -3, -4, -5, 1, 2, 3, 4, 5]) == (-1, 1)

# id_136_Test.py
from scenario1.id_136 import largest_smallest_integers

import pytest

class id_136_Test:
    """
    Test class of id_136.py.
    It contains ten unit test cases for the :func:`scenario1.id_136.largest_smallest_integers`.
    """
		
    def test_largest_smallest_integers_1(self):
        """
        Test case 1 for :func:`scenario1.id_136.largest_smallest_integers`.
        """
        assert largest_smallest_integers([-1, -2, -3, -4, -5]) == (-1, None)

    def test_largest_smallest_integers_2(self):
        """
        Test case 2 for :func:`scenario1.id_136.largest_smallest_integers`.
        """
        assert largest_smallest_integers([1, 2, 3, 4, 5]) == (None, 1)

    def test_largest_smallest_integers_3(self):
        """
        Test case 3 for :func:`scenario1.id_136.largest_smallest_integers`.
        """
        assert largest_smallest_integers([-1, 2, -3, 4, -5]) == (-1, 2)

    def test_largest_smallest_integers_4(self):
        """
        Test case 4 for :func:`scenario1.id_136.largest_smallest_integers`.
        """
        assert largest_smallest_integers([-1, -2, -3, -4, -5, 1, 2, 3, 4, 5]) == (-1, 1)

    def test_largest_smallest_integers_5(self):
        """
        Test case 5 for :func:`scenario1.id_136.largest_smallest_integers`.
        """
        assert largest_smallest_integers([-1, -2, -3, -4, -5, 1, 2, 3, 4, 5, -1, -2, -3, -4, -5, 1, 2, 3, 4, 5]) == (-1, 1)

    def test_largest_smallest_integers_6(self):
        """
        Test case 6 for :func:`scenario1.id_136.largest_smallest_integers`.
        """
        assert largest_smallest_integers([-1, -2, -3,

File: RQ2_Prompt_Elements/GPT3.5_Data/HumanEvalJava_output/scenario3/id_136T2000Test.java
// LargestSmallestIntegersTest.java
package scenario3;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
* Test class of {@link LargestSmallestIntegers}.
* It contains ten unit test cases for the {@link LargestSmallestIntegers#largestSmallestIntegers(List)} method.
*/
class LargestSmallestIntegersTest {

    @Test
    void testLargestSmallestIntegers1() {
        List<Object> lst = new ArrayList<>(Arrays.asList(2, 4, 1, 3, 5, 7));
        List<Integer> expected = new ArrayList<>(Arrays.asList(null, 1));
        assertEquals(expected, LargestSmallestIntegers.largestSmallestIntegers(lst));
    }

    @Test
    void testLargestSmallestIntegers2() {
        List<Object> lst = new ArrayList<>();
        List<Integer> expected = new ArrayList<>(Arrays.asList(null, null));
        assertEquals(expected, LargestSmallestIntegers.largestSmallestIntegers(lst));
    }

    @Test
    void testLargestSmallestIntegers3() {
        List<Object> lst = new ArrayList<>(Arrays.asList(0));
        List<Integer> expected = new ArrayList<>(Arrays.asList(null, null));
        assertEquals(expected, LargestSmallestIntegers.largestSmallestIntegers(lst));
    }

    @Test
    void testLargestSmallestIntegers4() {
        List<Object> lst = new ArrayList<>(Arrays.asList(2, 4, 1, 3, 5, 7, 0));
        List<Integer> expected = new ArrayList<>(Arrays.asList(null, 1));
        assertEquals(expected, LargestSmallestIntegers.largestSmallestIntegers(lst));
    }

    @Test
    void testLargestSmallestIntegers5() {
        List<Object> lst = new ArrayList<>(Arrays.asList(1, 3, 2, 4, 5, 6, -2));
        List<Integer> expected = new ArrayList<>(Arrays.asList(-2, 1));
        assertEquals(expected, LargestSmallestIntegers.largestSmallestIntegers(lst));
    }

    @Test
    void testLargestSmallestIntegers6() {
        List<Object> lst = new ArrayList<>(Arrays.asList(4, 5, 3, 6, 2, 7, -7));
        List<Integer> expected = new ArrayList<>(Arrays.asList(-7, 2));
        assertEquals(expected, LargestSmallestIntegers.largestSmallestIntegers(lst));
    }

    @Test
    void testLargestSmallestIntegers7() {
        List<Object> lst = new ArrayList<>(Arrays.asList(7, 3, 8, 4, 9, 2, 5, -9));
        List<Integer> expected = new ArrayList<>(Arrays.asList(-9, 2));
        assertEquals(expected, LargestSmallestIntegers.largestSmallestIntegers(lst));
    }

    @Test
    void testLargestSmallestIntegers8() {
        List<Object> lst = new ArrayList<>(Arrays.asList(-1, -3, -5, -6));
        List<Integer> expected = new ArrayList<>(Arrays.asList(-1, null));
        assertEquals(expected, LargestSmallestIntegers.largestSmallestIntegers(lst));
    }

    @Test
    void testLargestSmallestIntegers9() {
        List<Object> lst = new ArrayList<>(Arrays.asList(-1, -3, -5, -6, 0));
        List<Integer> expected = new ArrayList<>(Arrays.asList(-1, null));
        assertEquals(expected, LargestSmallestIntegers.largestSmallestIntegers(lst));
    }

    @Test
    void testLargestSmallestIntegers10() {
        List<Object> lst = new ArrayList<>(Arrays.asList(-6, -4, -4, -3, 1));
        List<Integer> expected = new ArrayList<>(Arrays.asList(-3, 1));
        assertEquals(expected, LargestSmallestIntegers.largestSmallestIntegers(lst));
    }

    @Test
    void testLargestSmallestIntegers11() {
        List<Object> lst = new ArrayList<>(Arrays.asList(-6, -4, -4, -3, -100, 1));
        List<Integer> expected = new ArrayList<>(Arrays.asList(-3, 1));
        assertEquals(expected, LargestSmallestIntegers.largestSmallestIntegers(lst));
    }
}